steps/previsit/first-attempt.tsx

```diff
@@ -0,0 +1,147 @@
+import { Heading } from "lbh-frontend-react";
+import { FieldsetLegend, HeadingLevels } from "lbh-frontend-react/components";
+import React from "react";
+import {
+  ComponentDatabaseMap,
+  ComponentWrapper,
+  DynamicComponent,
+  StaticComponent,
+} from "remultiform/component-wrapper";
+import { TransactionMode } from "remultiform/database";
+import { Checkboxes, CheckboxesProps } from "../../components/Checkboxes";
+import { makeSubmit } from "../../components/makeSubmit";
+import { TextArea } from "../../components/TextArea";
+import keyFromSlug from "../../helpers/keyFromSlug";
+import ProcessDatabaseSchema from "../../storage/ProcessDatabaseSchema";
+import Storage from "../../storage/Storage";
+import PageSlugs from "../PageSlugs";
+import PageTitles from "../PageTitles";
+
+const storeFirstVisitDate = async (): Promise<void> => {
+  const db = await Storage.ProcessContext?.database;
+  const processRef = keyFromSlug()();
+  const date = new Date().toISOString();
+  if (db) {
```

Comment:
I'd switch this to be a guard statement immediately after checking the DB. The flatter the code, the better. Also `console.error` something if it's falsy, as that should never be the case when calling this function.

Reply:
I'm not sure what you mean here but I'll get George to look at it with me maybe.

Reply:
I mean reverse the logic, so instead of `if (db) { /* do something with the db */ }` you do `if (!db) { return } /* do something with the db */`.